core/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/StringTerms.java

```diff
@@ -67,6 +67,11 @@ public Object getKey() {
         @Override
         public Number getKeyAsNumber() {
             // this method is needed for scripted numeric aggs
+            try {
+                return Long.parseLong(termBytes.utf8ToString());
+            } catch (NumberFormatException ignored) {
+                //failed to detect an Integer
```

Comment:
@matarrese This is still not what we would want in a comment here. We want to explain why we doing this. Please look at the original issue that provoked this change and leave a comment here that explains the why behind first parsing as a long, and then leniently parsing as a double.